### PR TITLE
adds support for email as username

### PR DESCRIPTION
**Pull request by staranjeet**
This PR adds support for using email as username in the urls.

### CODE REVIEW

**Comment by kracekumar** [comment]
Username is of  30 characters max, email can be upto 255 characters.

On Mon, Nov 16, 2015 at 5:21 PM, Taranjeet Singh <notifications@github.com>
wrote:

> This PR adds support for using email as username in the urls.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/pythonindia/wye/pull/190
> Commit Summary
>
>    - adds support for email as username
>
> File Changes
>
>    - *M* wye/profiles/urls.py
>    <https://github.com/pythonindia/wye/pull/190/files#diff-0> (4)
>
> Patch Links:
>
>    - https://github.com/pythonindia/wye/pull/190.patch
>    - https://github.com/pythonindia/wye/pull/190.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pythonindia/wye/pull/190>.
>



-- 
Regards
Kracekumar Ramaraj
http://kracekumar.com

**Comment by vnbang2003** [comment]
@kracekumar I agree but we can allow special character in slug name right ? The problem is special character @ and .

**Comment by kracekumar** [comment]
`Django` takes care of sanitizing the values. Only concern is postgres will throw up when username is more than 30 characters. If that is taken care please avoid the comment.

**Comment by vnbang2003** [comment]
@staranjeet Can you confirm if @kracekumar is been taken care in our code?

**Comment by staranjeet** [comment]
I tried adding email as username with more than 30 characters, but the input box is taking only first 30 characters. The profile got created with those 30 characters as username, so I think since django is taking care of this validation, postgres should not come in picture. I have tested this on my local.

**Comment by vnbang2003** [comment]
@staranjeet you mean django truncated character after 30 ? In that case I think we need to show proper message and ask user to reduce character right ?

**Comment by staranjeet** [comment]
yeah indirectly. The input field that was generated by form is
`<input autofocus="autofocus" id="id_username" maxlength="30" name="username" placeholder="Username" type="text" />`.

So this is django + html which is taking care of that.

**Comment by vnbang2003** [comment]
@staranjeet Cool that address all issue.
@kracekumar Did it resolve all you concern ? If Yes can you accept pull request ?

**Comment by kracekumar** [comment]
Vijay you can merge it. No worries :-)

On Tue, Nov 17, 2015 at 9:36 PM, Vijay <notifications@github.com> wrote:

> @staranjeet <https://github.com/staranjeet> Cool that address all issue.
> @kracekumar <https://github.com/kracekumar> Did it resolve all you
> concern ? If Yes can you accept pull request ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pythonindia/wye/pull/190#issuecomment-157415097>.
>



-- 
Regards
Kracekumar Ramaraj
http://kracekumar.com

File: wye/profiles/urls.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 urlpatterns = [
     url(r'^dashboard$', views.UserDashboard.as_view(),
         name="dashboard"),
-    url(r'^(?P<slug>[a-zA-Z0-9]+)/$',
+    url(r'^(?P<slug>[a-zA-Z0-9@.]+)/$',
         views.ProfileView.as_view(), name='profile-page'),
-    url(r'^(?P<slug>[a-zA-Z0-9]+)/edit$',
+    url(r'^(?P<slug>[a-zA-Z0-9@.]+)/edit$',
```
Comment (theskumar) [inline wye/profiles/urls.py]: @staranjeet @vnbang2003 Emails can also have `+` sign in them. 
Also I think using email in url is good idea. I would suggest using the user_id in the profile url if username is not present.

Comment (theskumar) [inline wye/profiles/urls.py]: s/good idea/not a good idea/

Comment (theskumar) [inline wye/profiles/urls.py]: Ah. didn't notice that subject of PR is not what the code does. The code is fine given the conversation.

**Pull request merged**